backend/infrahub/api/oidc.py

```diff
@@ -185,3 +188,28 @@ def _validate_response(response: httpx.Response) -> None:
         body=response.json(),
     )
     raise GatewayError(message="Invalid response from Authentication provider")
+
+
+async def _get_id_token_groups(
```

Comment:
Would be good to add a unit test for this function with the expected format of the token

Reply:
I've added some tests around this function now.